Tests/PackageLoadingTests/PkgConfigTests.swift

```diff
@@ -48,52 +48,61 @@ class PkgConfigTests: XCTestCase {
+                switch result.provider {
+                case .brew(let names)?:
+                    XCTAssertEqual(names, ["libFoo"])
+                case .apt(let names)?:
+                    XCTAssertEqual(names, ["libFoo-dev"])
+                case .yum(let names)?:
+                    XCTAssertEqual(names, ["libFoo-devel"])
+                case nil:
+                    XCTFail("Expected a provider here")
+                }
+                XCTAssertTrue(result.couldNotFindConfigFile)
+                switch result.error {
                 case PkgConfigError.couldNotFindConfigFile?: break
                 default:
-                XCTFail("Unexpected error \(String(describing: result.error))")
+                    XCTFail("Unexpected error \(String(describing: result.error))")
+                }
             }
         }
 
         // Pc file.
         try withCustomEnv(["PKG_CONFIG_PATH": inputsDir.pathString]) {
-            let result = pkgConfigArgs(for: SystemLibraryTarget(pkgConfig: "Foo"), diagnostics: diagnostics)!
-            XCTAssertEqual(result.pkgConfigName, "Foo")
-            XCTAssertEqual(result.cFlags, ["-I/path/to/inc", "-I\(inputsDir.pathString)"])
-            XCTAssertEqual(result.libs, ["-L/usr/da/lib", "-lSystemModule", "-lok"])
-            XCTAssertNil(result.provider)
-            XCTAssertNil(result.error)
-            XCTAssertFalse(result.couldNotFindConfigFile)
+            let results = pkgConfigArgs(for: SystemLibraryTarget(pkgConfig: "Foo"), diagnostics: diagnostics)
+            for wrappedResult in results {
+                let result = wrappedResult!
+                XCTAssertEqual(result.pkgConfigName, "Foo")
+                XCTAssertEqual(result.cFlags, ["-I/path/to/inc", "-I\(inputsDir.pathString)"])
+                XCTAssertEqual(result.libs, ["-L/usr/da/lib", "-lSystemModule", "-lok"])
+                XCTAssertNil(result.provider)
+                XCTAssertNil(result.error)
+                XCTAssertFalse(result.couldNotFindConfigFile)
+            }
         }
 
         // Pc file with prohibited flags.
         try withCustomEnv(["PKG_CONFIG_PATH": inputsDir.pathString]) {
-            let result = pkgConfigArgs(for: SystemLibraryTarget(pkgConfig: "Bar"), diagnostics: diagnostics)!
-            XCTAssertEqual(result.pkgConfigName, "Bar")
-            XCTAssertEqual(result.cFlags, ["-I/path/to/inc"])
-            XCTAssertEqual(result.libs, ["-L/usr/da/lib", "-lSystemModule", "-lok"])
-            XCTAssertNil(result.provider)
-            XCTAssertFalse(result.couldNotFindConfigFile)
-            switch result.error {
-            case PkgConfigError.prohibitedFlags(let desc)?:
-                XCTAssertEqual(desc, "-DDenyListed")
-            default:
-                XCTFail("unexpected error \(result.error.debugDescription)")
+            let results = pkgConfigArgs(for: SystemLibraryTarget(pkgConfig: "Bar"), diagnostics: diagnostics)
+            for wrappedResult in results {
```

Comment:
The unwrapping here shouldn't be needed anymore, since the results aren't optional now.